### ISSUE TITLE
[Bug] An error during the "sendBase" phase in the thrift plugin resulted in a duplicated traceId

### ISSUE THREAD

**Issue by zhaoxiaojie0415** [apache/skywalking#10438]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar issues.

### Apache SkyWalking Component

Java Agent (apache/skywalking-java)

### What happened

I use the skywalking v8.14.0 and thrift 0.9.3. In my case, A service communicate B service via thrift. If an error has occurred during the sendBase phase,  the ExitSpan would not be stopped。When the next request comes, the same thread always repeats the traceid generated for the first time.

![企业微信截图_21cfb01b-ec7b-4db4-9b3d-6766bf31b47e](https://user-images.githubusercontent.com/13363629/220853756-01ad9f57-a160-4e26-85ba-8edd1e01730f.png)
![WX20230223-161345](https://user-images.githubusercontent.com/13363629/220853794-9d6d4095-2302-4d83-9b78-28d16ad1a7b1.png)

### What you expected to happen

TServiceClientInterceptor does not stop Span. It seems that skywalking only stop the ExitSpan generated by thrift after the ”receiveBase“ method in TServiceClientReceiveInterceptor class.

### How to reproduce

We can define a thrift IDL as follows:
service TestThrift {
struct TRequest{
   1: map<string, string> property;
}

void testRpc(1: TRequest request)
}

We can create a TRequest, then put  "key="test", value=null" into the property. This would be generated a NPE in the "sendBase" phase. Subsequently, we keep making RPC requests, and we can observe that the same thread is consistently using the same traceId.

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by wu-sheng** [comment]
TBO, thrift was not mentioned once after the first adding. I don't have much to share, and as a plugin, you need to debug and find out what is happening.

This is what I can share
- This is the demo application we used for testing, https://github.com/apache/skywalking-java/tree/main/test/plugin/scenarios/thrift-scenario. It is verified.
- These are tested versions, https://github.com/apache/skywalking-java/blob/main/test/plugin/scenarios/thrift-scenario/support-version.list

Basically, if your codes are out of this scope, we are not tested.

**Comment by wu-sheng** [comment]
> TServiceClientInterceptor does not stop Span. It seems that skywalking only stop the ExitSpan generated by thrift after the ”receiveBase“ method in TServiceClientReceiveInterceptor class.

If that is so, you should find another suitable place to close the span instead, when error happens.

**Comment by zhaoxiaojie0415** [comment]
> * https://github.com/apache/skywalking-java/tree/main/test/plugin/scenarios/thrift-scenario



> TBO, thrift was not mentioned once after the first adding. I don't have much to share, and as a plugin, you need to debug and find out what is happening.
> 
> This is what I can share
> 
> * This is the demo application we used for testing, https://github.com/apache/skywalking-java/tree/main/test/plugin/scenarios/thrift-scenario. It is verified.
> * These are tested versions, https://github.com/apache/skywalking-java/blob/main/test/plugin/scenarios/thrift-scenario/support-version.list
> 
> Basically, if your codes are out of this scope, we are not tested.



> TBO, thrift was not mentioned once after the first adding. I don't have much to share, and as a plugin, you need to debug and find out what is happening.
> 
> This is what I can share
> 
> * This is the demo application we used for testing, https://github.com/apache/skywalking-java/tree/main/test/plugin/scenarios/thrift-scenario. It is verified.
> * These are tested versions, https://github.com/apache/skywalking-java/blob/main/test/plugin/scenarios/thrift-scenario/support-version.list
> 
> Basically, if your codes are out of this scope, we are not tested.

<img width="1003" alt="WX20230223-193724@2x" src="https://user-images.githubusercontent.com/13363629/220895900-fc82b3bb-5583-42bd-bd3b-2de0615f783a.png">

Thanks! In this test demo, after thrift compile, the method "echo" has two phase "send_echo" and "recv_echo".
The current thrift plugin indeed create span before "send_echo", and stop span after "recv_echo", without considering error situation.

I think it need also to stop span when the sendBase error, i.e. stop span in the method "handleMethodException" of "TServiceClientInterceptor".

If need, I can  submit an PR later and try to solve this issue.
By the way, this plugin can also support thrift v0.9.3， which can be added in the support-version list.

**Comment by wu-sheng** [comment]
Yes, I think it makes sense you submit a pull request to cover both.

**Comment by zhaoxiaojie0415** [comment]
#https://github.com/apache/skywalking-java/pull/467
